src/util/helpers.ts

```diff
@@ -1,14 +1,33 @@
-import { ConflictException, NotFoundException } from '@nestjs/common'
+import { ConflictException } from '@nestjs/common'
 import { UserWorkspace } from 'src/workspaces/entities/user-workspace.entity'
+import { Client } from 'src/clients/entities/client.entity'
+import { Project } from 'src/projects/entities/project.entity'
 export function verifyIfNameNotTaken(userWorkspace?: UserWorkspace) {
   if (!userWorkspace || !userWorkspace.workspace) return
 
   const {
     workspace: { name: existingName },
   } = userWorkspace
+
   if (existingName) {
     throw new ConflictException(
       `A workspace with the name ${existingName} already exists`,
     )
```

Comment:
Please convert this function into service function across their respective services and make sure to reuse them across different operations such as create or update